src/main/java/edu/harvard/iq/dataverse/GlobalId.java

```diff
@@ -119,9 +149,11 @@ public URL toURL() {
      *       identifier: 111012
      *
      * @param identifierString
-     * 
+     * @param separator the string that separates the authority from the identifier.
+     * @param destination the global id that will contain the parsed data.
+     * @return {@code destination}, after its fields have been updated, or
+     *         {@code null} if parsing failed.
      */
-    
     private boolean parsePersistentId(String identifierString) {
```

Comment:
As part of #4761 we have been discussing pid structure and the challenges of external systems using these values. There is concern that we will start importing pids with a lot of weird characters, for example this valid DOI:

`https://doi.org/10.1002/(sici)1099-1409(199908/10)3:6/7<672::aid-jpp192>3.0.co;2-8`

This is of more concern as we are trying to have all external systems use pids to reference files and datasets, but these edge case pids could lead to systems breaking, injection attacks, etc.

A solution that was discussed was to enforce limits on the characters we import. The general idea is to be pretty strict but simple and then add in edge cases as requested. 

See here for more info on DataCite's pid. I did not read into Handle's or ezid, tho my limited understanding is that those allow a different/weird assortment of characters. https://support.datacite.org/docs/doi-basics https://www.crossref.org/blog/dois-and-matching-regular-expressions/

Reply:
@michbarsinai I'm worried about "opening the floodgates" and allowing in pids well outside our normal structure. My understanding is that before this PR it was impossible to import DOIs in. It would be much better to have the restrictions from the get-go. 

Maybe this could be done in another issue before the next release as the work is small and fairly isolated. @scolapasta thoughts?

Reply:
Yes, let's keep it separate. @matthew-a-dunlap can you open an issue? We can convince @djbrooke of its priority next week when he's back.

Reply:
@scolapasta will do before I head out!

Reply:
I defer to @scolapasta. Here are my thought on this: We are parsing the PID using `GlobalId.parse` (https://github.com/IQSS/dataverse/blob/3083-support-migration-with-dois/src/main/java/edu/harvard/iq/dataverse/api/Dataverses.java#L257)
So changes there will be used in the new code here. Also, this is a large PR (131 files) so any delay here causes a lot of merge and delete/edit conflicts. So I think implementing #4761 in a new PR would make more sense.